### ISSUE TITLE
ubuntu17.04 ansible: issue with ReaR snapshot not signed

### ISSUE THREAD

**Issue by gdha** [gdha/rear-automated-testing#29]
The question is how can we have it signed at OBS?

````
TASK [rear-test : install aptitude (need this for upgrade action)] *************
changed: [server]

TASK [rear-test : update all packages] *****************************************
changed: [server]

TASK [rear-test : add http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_ansible_distribution_version repo] ***
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: apt.cache.FetchFailedException: W:GPG error: http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 6B7485DB725A0C43, E:The repository 'http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release' is not signed.
fatal: [server]: FAILED! => {"changed": false, "failed": true, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_vyjIeU/ansible_module_apt_repository.py\", line 565, in <module>\n    main()\n  File \"/tmp/ansible_vyjIeU/ansible_module_apt_repository.py\", line 553, in main\n    cache.update()\n  File \"/usr/lib/python2.7/dist-packages/apt/cache.py\", line 464, in update\n    raise FetchFailedException(e)\napt.cache.FetchFailedException: W:GPG error: http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 6B7485DB725A0C43, E:The repository 'http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release' is not signed.\n", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 0}
````

**Comment by gdha** [comment]
Trying it manually:
````
root@server:/home/vagrant# add-apt-repository http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04/ 

root@server:/home/vagrant# apt-get update
Hit:1 http://ppa.launchpad.net/ansible/ansible/ubuntu zesty InRelease
Get:2 http://security.ubuntu.com/ubuntu zesty-security InRelease [89.2 kB]                                                        
Hit:3 http://archive.ubuntu.com/ubuntu zesty InRelease                                                           
Get:4 http://archive.ubuntu.com/ubuntu zesty-updates InRelease [89.2 kB]                                                           
Ign:5 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty InRelease
Ign:6 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ InRelease
Ign:7 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty Release
Get:8 http://archive.ubuntu.com/ubuntu zesty-backports InRelease [89.2 kB]
Get:9 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release [1,046 B]
Ign:10 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main amd64 Packages
Ign:11 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main all Packages
Ign:12 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main i386 Packages
Ign:13 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main Translation-en
Ign:14 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main Translation-en_US
Get:15 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release.gpg [189 B]
Ign:15 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release.gpg
Ign:10 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main amd64 Packages
Ign:11 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main all Packages
Ign:12 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main i386 Packages
Ign:13 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main Translation-en
Ign:14 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main Translation-en_US
Get:16 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Packages [659 B]
Ign:10 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main amd64 Packages
Ign:11 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main all Packages
Ign:12 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main i386 Packages
Ign:13 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main Translation-en
Ign:14 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main Translation-en_US
Ign:10 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main amd64 Packages
Ign:11 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main all Packages
Ign:12 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main i386 Packages
Ign:13 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main Translation-en
Ign:14 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main Translation-en_US
Ign:10 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main amd64 Packages
Ign:11 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main all Packages
Ign:12 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main i386 Packages
Ign:13 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main Translation-en
Ign:14 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main Translation-en_US
Err:10 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main amd64 Packages
  404  Not Found
Ign:11 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main all Packages
Ign:12 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main i386 Packages
Ign:13 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main Translation-en
Ign:14 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty/main Translation-en_US
Fetched 270 kB in 4s (54.8 kB/s)
Reading package lists... Done
W: The repository 'http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty Release' does not have a Release file.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.
W: GPG error: http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 6B7485DB725A0C43
W: The repository 'http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release' is not signed.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: Failed to fetch http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04/dists/zesty/main/binary-amd64/Packages  404  Not Found
E: Some index files failed to download. They have been ignored, or old ones used instead.
````
Getting the exact same error.

**Comment by gdha** [comment]
Today we were able to fix on OBS the rear unstable versions for Debian/Ubuntu, so we retry it:
````
root@server:/home/vagrant# apt update
Hit:1 http://ppa.launchpad.net/ansible/ansible/ubuntu zesty InRelease
Hit:2 http://archive.ubuntu.com/ubuntu zesty InRelease                                           
Ign:3 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty InRelease
Hit:4 http://archive.ubuntu.com/ubuntu zesty-updates InRelease                                   
Ign:5 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ InRelease                
Hit:6 http://archive.ubuntu.com/ubuntu zesty-backports InRelease                                                            
Err:7 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty Release
  404  Not Found
Get:8 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release [1,046 B]
Get:9 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release.gpg [189 B]
Hit:10 http://security.ubuntu.com/ubuntu zesty-security InRelease       
Ign:9 http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release.gpg
Reading package lists... Done  
E: The repository 'http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 zesty Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
W: GPG error: http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 6B7485DB725A0C43
E: The repository 'http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
````
However, the Release and Release.gpg files are present on OBS. Not sure what goes wrong - perhaps my path to rear?

**Comment by gdha** [comment]
Oh boy though issue as now I get:
````
TASK [rear-test : download the Release.key from OBS] ***************************
changed: [server]

TASK [rear-test : add http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_version repo] ***
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: apt.cache.FetchFailedException: W:GPG error: http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release: The following signatures were invalid: F532523ADE4BBF1CBFF6523F6B7485DB725A0C43, E:The repository 'http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release' is not signed.
fatal: [server]: FAILED! => {"changed": false, "failed": true, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_cSbGDk/ansible_module_apt_repository.py\", line 565, in <module>\n    main()\n  File \"/tmp/ansible_cSbGDk/ansible_module_apt_repository.py\", line 553, in main\n    cache.update()\n  File \"/usr/lib/python2.7/dist-packages/apt/cache.py\", line 464, in update\n    raise FetchFailedException(e)\napt.cache.FetchFailedException: W:GPG error: http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release: The following signatures were invalid: F532523ADE4BBF1CBFF6523F6B7485DB725A0C43, E:The repository 'http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release' is not signed.\n", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 0}
````

I downloaded the Release files from OBS and tried to import them manually:
````
root@server:/etc/apt/trusted.gpg.d# file Release*
Release:     ASCII text
Release.gpg: PGP signature Signature (old)
Release.key: PGP public key block Public-Key (old)
root@server:/etc/apt/trusted.gpg.d# apt-key add Release
gpg: no valid OpenPGP data found.
root@server:/etc/apt/trusted.gpg.d# apt-key add Release.key
gpg: [don't know]: invalid packet (ctb=2d)
gpg: keyring_get_keyblock: read error: Invalid packet
gpg: error reading keyblock: Invalid keyring
gpg: key export failed: Invalid keyring
gpg: [don't know]: invalid packet (ctb=2d)
gpg: keyring_get_keyblock: read error: Invalid packet
gpg: error reading keyblock: Invalid keyring
gpg: key export failed: Invalid keyring
gpg: [don't know]: invalid packet (ctb=2d)
gpg: keyring_get_keyblock: read error: Invalid packet
gpg: error reading keyblock: Invalid keyring
gpg: key export failed: Invalid keyring
gpg: [don't know]: invalid packet (ctb=2d)
gpg: keydb_get_keyblock failed: Value not found
gpg: Total number processed: 1
gpg:       skipped new keys: 1
root@server:/etc/apt/trusted.gpg.d# apt-key add Release.gpg
gpg: no valid OpenPGP data found.
````
As you can see they are not valid - so the issue is on the OBS side - how do we generate a valid repo key?

**Comment by gdha** [comment]
By creating file:
````
# cat /etc/apt/apt.conf.d/99allowinsecurerepositories
Acquire::AllowInsecureRepositories "true";
````
the `apt update` produces only warnings, but it is ugly:
````
W: GPG error: http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 6B7485DB725A0C43
W: The repository 'http://download.opensuse.org/repositories/Archiving:/Backup:/Rear:/Snapshot/xUbuntu_17.04 ./ Release' is not signed.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.
````
Perhaps, it is a good idea to log an issue at https://github.com/openSUSE/obs-build
